ikarus/finiteelements/mechanics/materials/vanishingstress.hh

```diff
@@ -170,12 +170,12 @@ private:
 
     auto f = [&](auto&) {
       auto S = matImpl_.template stresses<Underlying::strainTag, true>(E);
-      S.array() += tol_;
+      S.array() += 1e-14; // to circumvent division-by-zero in NewtonRaphson method while using AutoDiff when rx=0
```

Comment:
Now you perturb all the time for all iterations. I think it would be better only for the first one.

Maybe you can do something like: https://godbolt.org/z/Yb95WzzMq

Also I think it is enough to perturb only the stresses and not the tangent moduli.

Reply:
Is this still needed, as it doesn't solve the issue at hand (see [comment](https://github.com/ikarus-project/ikarus/pull/329#issuecomment-2416967885)). May be this can be investigated in detail and fixed everywhere while focusing on Issue #330 and skip it for this PR, as the CI works?